drivers/amazonec2/amazonec2.go

```diff
@@ -1034,22 +1034,23 @@ func (d *Driver) configureSecurityGroups(groupNames []string) error {
 			}
 		}
 		d.SecurityGroupIds = append(d.SecurityGroupIds, *group.GroupId)
+	}
+
+	var mainGroup = groupsByName[groupNames[0]]
+	perms, err := d.configureSecurityGroupPermissions(mainGroup)
+	if err != nil {
+		return err
+	}
 
-		perms, err := d.configureSecurityGroupPermissions(group)
+	if len(perms) != 0 {
+		log.Debugf("authorizing group %s with permissions: %v", groupNames, perms)
+		_, err := d.getClient().AuthorizeSecurityGroupIngress(&ec2.AuthorizeSecurityGroupIngressInput{
+			GroupId:       mainGroup.GroupId,
+			IpPermissions: perms,
+		})
 		if err != nil {
 			return err
 		}
-
-		if len(perms) != 0 {
-			log.Debugf("authorizing group %s with permissions: %v", groupNames, perms)
```

Comment:
Mmh, just realized that this debug message was probably incorrect. We should log the group name that is being set and not all group names.